sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java

```diff
@@ -367,10 +375,21 @@ private synchronized void checkPushInIntent(Activity activity, Intent intent) {
                 if (mFirstGoogleMessageId == null) {
                     mFirstGoogleMessageId = googleMessageId;
                 }
-                PushNotification notification = new PushNotification(intent);
+
                 AppCenterLog.info(LOG_TAG, "Clicked push message from background id=" + googleMessageId);
                 mLastGoogleMessageId = googleMessageId;
                 AppCenterLog.debug(LOG_TAG, "Push intent extras=" + intent.getExtras());
+
+                Intent currentPushIntent = intent;
+                for (Intent receivedIntent : mReceivedIntents) {
+                    if (googleMessageId.equals(PushIntentUtils.getMessageId(receivedIntent))) {
```

Comment:
Switch to map as mentioned in the other comment.
Also please add a comment every time there is a newline to start a new code block.